### PR TITLE
build: webpack timeline loader

### PR DESCRIPTION
**Pull request by MaikoTan**
A too-young-too-simple timeline loader who remove comments and blank lines only.

### CODE REVIEW

File: webpack/loaders/timeline-loader.cjs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+module.exports = function(content, map, meta) {
+  let ret = '';
+
+  content.split(/\r?\n/).forEach((_line) => {
+    let line = _line.trim();
+    if (!line || line.startsWith('#'))
+      return;
+    line = line.replace(/#.*$/, '').trim();
+    ret += line + '\r\n';
```
Comment (quisquous) [inline webpack/loaders/timeline-loader.cjs:11]: What about removing lines that are only comments but `#` doesn't start at the beginning of the line too?

```suggestion
    let line = _line.replace(/#.*$/, '').trim();
    if (!line)
      return;
    ret += line + '\r\n';
```

But more importantly, either of these will break `100.0 "Some ability #1" sync /stuff/`.  Can you fix the regex to handle # inside of quotes as well?

Comment (MaikoTan) [reply webpack/loaders/timeline-loader.cjs:11]: Oh, yes.
I gonna fix it soon.

Comment (MaikoTan) [reply webpack/loaders/timeline-loader.cjs:11]: How about this regex?
https://regex101.com/r/CkKpW6/1

The only issue is when comment contain one quote (") would causing regex unmatched.

Comment (quisquous) [reply webpack/loaders/timeline-loader.cjs:11]: https://regex101.com/r/FwTpMe/2 maybe?

Comment (MaikoTan) [reply webpack/loaders/timeline-loader.cjs:11]: lookbehind sounds good, but is it possible that '#' in `sync /regex/` things?

Comment (quisquous) [reply webpack/loaders/timeline-loader.cjs:11]: Oh, that's quite right!

Comment (MaikoTan) [reply webpack/loaders/timeline-loader.cjs:11]: Looks like javascript regex doesn't offer something like [`(?(exp)yes|no)`](https://docs.microsoft.com/en-gb/dotnet/standard/base-types/regular-expression-language-quick-reference?redirectedfrom=MSDN#alternation-constructs), so don't know how to capture the `sync /regex/` part as well.

Comment (quisquous) [reply webpack/loaders/timeline-loader.cjs]: Hmm, how about: https://regex101.com/r/FwTpMe/3

Comment (MaikoTan) [reply webpack/loaders/timeline-loader.cjs]: I tried several times and it works well, thanks!

Comment (quisquous) [reply webpack/loaders/timeline-loader.cjs]: Wait sorry: https://regex101.com/r/FwTpMe/4

One character change to support slashes in the ability name!

Comment (quisquous) [reply webpack/loaders/timeline-loader.cjs]: Haha, I see that you literally just updated this as I had this thought.

**Pull request merged**